homeassistant/components/netatmo/sensor.py

```diff
@@ -547,46 +517,93 @@ def unit_of_measurement(self):
         """Return the unit of measurement of this entity."""
         return self._unit_of_measurement
 
-    @property
-    def unique_id(self):
-        """Return the unique ID for this sensor."""
-        return self._unique_id
-
     @property
     def available(self):
         """Return True if entity is available."""
         return self._state is not None
 
-    def update(self):
-        """Get the latest data from Netatmo API and updates the states."""
-        self.netatmo_data.update()
+    @property
+    def _data(self):
+        return self.data_handler.data[self._signal_name]
+
+    async def async_added_to_hass(self) -> None:
+        """Entity created."""
+        await super().async_added_to_hass()
+
+        self.data_handler.listeners.append(
+            async_dispatcher_connect(
```

Comment:
Can we get rid of the dispatcher here?

Reply:
As discussed on Discord, this wont be changed easily and can be left for a future PR.